lib/ex_unit/lib/ex_unit/diff.ex

```diff
@@ -377,7 +377,13 @@ defmodule ExUnit.Diff do
     {parsed_left, improper_left, operators_left, length_left} =
       split_left_list(left, 0, env.context)
 
-    {parsed_right, improper_right} = split_right_list(right, length_left, [])
+    {parsed_right, improper_right} =
+      if improper_left != [] do
+        split_right_list(right, length_left, [])
+      else
+        split_right_list(right, [])
```

Comment:
maybe there is a better way - passing a big number as length_left works here too but is ugly